src/index.js

```diff
@@ -77,46 +82,28 @@ class DelegatedContentRouting {
 
     options.maxTimeout = options.maxTimeout || DEFAULT_MAX_TIMEOUT
 
-    this.dht.findProvs(key.toString(), {
-      timeout: `${options.maxTimeout}ms` // The api requires specification of the time unit (s/ms)
-    }, callback)
+    this._httpQueue.add(() =>
+      this.dht.findProvs(key.toString(), {
+        timeout: `${options.maxTimeout}ms` // The api requires specification of the time unit (s/ms)
+      }, callback)
+    ).catch(callback)
   }
 
   /**
    * Announce to the network that the delegated node can provide the given key.
    *
    * Currently this uses the following hack
-   * - call swarm.connect on the delegated node to us, to ensure we are connected
-   * - call refs --recursive on the delegated node, so it fetches the content
+   * - delegate is one of bootstrap nodes, so we are always connected to it
+   * - call refs on the delegated node, so it fetches the content
    *
    * @param {CID} key
    * @param {function(Error)} callback
    * @returns {void}
    */
   provide (key, callback) {
-    const addrs = this.bootstrappers.map((addr) => {
-      return addr.encapsulate(`/p2p-circuit/ipfs/${this.peerId.toB58String()}`)
-    })
-
-    series([
-      (cb) => parallel(addrs.map((addr) => {
-        return reflect((cb) => this.swarm.connect(addr.toString(), cb))
-      }), (err, results) => {
-        if (err) {
-          return cb(err)
-        }
-
-        // only some need to succeed
-        const success = results.filter((res) => res.error == null)
-        if (success.length === 0) {
-          return cb(new Error('unable to swarm.connect using p2p-circuit'))
-        }
-        cb()
-      }),
-      (cb) => {
-        this.refs(key.toString(), { recursive: true }, cb)
-      }
-    ], (err) => callback(err))
+    this._httpQueueRefs.add(() =>
+      this.refs(key.toString(), { recursive: false }, (err, res) => callback(err))
+    ).catch(callback)
```

Comment:
Same here!

Reply:
That's odd: last test (`provide`) stop passing if I move callback call to the outside:

```js
    this._httpQueueRefs.add(() =>
      this.refs(key.toString(), { recursive: false })
    ).then(res => callback(null, res), callback)
```

makes tests hang:

```
  6 passing (22s)
  1 failing

  1) DelegatedContentRouting
       provide
         should be able to register as a content provider to the delegate node:
     Error: Timeout of 20000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/lidel/project/js-libp2p-delegated-content-routing/test/index.spec.js)

```

Bit late on my end, so I can't tell if it is a bug somewhere, or just me being tired.
Thoughts?